crates/ruff/src/rule_selector.rs

```diff
@@ -167,9 +188,18 @@ impl IntoIterator for &RuleSelector {
                     .chain(Linter::Flake8Print.rules()),
             ),
             RuleSelector::Linter(linter) => RuleSelectorIter::Vec(linter.rules()),
-            RuleSelector::Prefix { prefix, .. } => RuleSelectorIter::Vec(prefix.clone().rules()),
+            RuleSelector::Prefix { prefix, .. } | RuleSelector::Rule { prefix, .. } => {
+                RuleSelectorIter::Vec(prefix.clone().rules())
+            }
         }
     }
+
+    /// Returns rules matching the selector, taking into account whether preview mode is enabled.
+    pub fn rules(&self, preview: PreviewMode) -> impl Iterator<Item = Rule> + '_ {
+        self.all_rules().filter(move |rule| {
+            matches!(self, RuleSelector::Rule { .. }) || preview.is_enabled() || !rule.is_preview()
```

Comment:
(If we want to avoid enabling preview selection for specific rules, then we'd want to remove `RuleSelector::Rule { .. })` here.)